src/bun.js/api/bun/socket.zig

```diff
@@ -2305,7 +2305,7 @@ fn NewSocket(comptime ssl: bool) type {
             const length_value = args[2];
 
             if (encoding_value != .undefined and (offset_value != .undefined or length_value != .undefined)) {
-                globalObject.throwTODO("Support encoding with offset and length altogether. Only either encoding or offset, length is supported, but not both combinations yet.");
+                globalObject.throwTODO("Support encoding with offset and length altogether. Only either encoding or offset, length is supported, but not both combinations yet.") catch {};
```

Comment:
Shouldn't we change `writeOrEnd` to have an error union return type? Instead of `catch {};`? Same with stdio.zig `extract`?

Reply:
perhaps, yeah it seems this whole function is using `.fail` for exceptions
@cirospaciari would likely know more on the prospect of that change